part-spec/power.json

```diff
@@ -697,6 +697,516 @@
+        },
+        "dimmingRatioTyp": {
+          "description": "typical dimming ratio of a device",
+          "type": "object",
+          "example": [
+          {"numerator": 20},
+          {"denominator": 1}
+          ]
+        },
+        "dimmingRatioMin": {
+          "description": "minimum dimming ratio of a device",
+          "type": "object",
+          "example": [
+          {"numerator": 10},
+          {"denominator": 1}
+          ]
+        },
+        "dimmingRatioMax": {
+          "description": "maximum dimming ratio of a device",
+          "type": "object",
+          "example": [
+          {"numerator": 40},
+          {"denominator": 1}
+          ]
+        },
+        "componentProtectionThresholds": {
+          "description": "Thermal and power supply protection thresholds of a device",
+          "type": "array",
+          "items": {
+            "$ref": "icProtection.json#/componentProtectionThresholds"
+          }
+        },
+        "package": {
+          "description": "package size of resistor",
+          "$ref": "definitions.json#/package"
+        },
+        "pins": {
+          "description": "array of pin objects with associated properties",
+          "type": "array",
+          "items": {
+            "$ref": "definitions.json#/pinSpec"
+          }
+        }
+      }
+    },
+    {
+      "title": "battery charger",
+      "type": "object",
+      "required": [
+        "componentID",
+        "batteryChargeCurrentMax"
+      ],
+      "properties": {
+        "componentID": {
+          "description": "common component identifying information, such as mpn",
+          "$ref": "definitions.json#/componentID"
+        },
+        "chargerType": {
+          "description": "battery charger type",
+          "examples": [
+            "linear",
+            "switching",
+            "direct"
+          ],
+          "type": "string"
+        },
+        "converterType": {
+          "description": "switching charger type",
+          "examples": [
+            "buck",
+            "boost",
+            "buck-boost",
+            "other"
+          ],
+          "type": "string"
+        },
+        "chargerTopology": {
+          "description": "type of battery charger topology (Narrow VDC vs Hybrid Power Boost)",
+          "enum": [
+            "nvdc",
+            "hpb"
+          ],
+          "type": "string"
+        },
+        "batteryConfig": {
+          "description": "battery configuration supported by the device",
+          "examples": [
+            "1S",
+            "2S2P",
+            "3S",
+            "4S",
+            "4S2P"
+          ],
+          "type": "string"
+        },
+        "batteryCellChemistry": {
+          "description": "battery cell chemistry supported by the device",
+          "examples": [
+            "Li-Ion",
+            "LiPo"
+          ],
+          "type": "string"
+        },
+        "inputPowerSource": {
```

Comment:
Would this just be more of a voltage range + some additional info if USB PD protocol is supported?

Reply:
This is more about the type of inputs supported: e.g: AC adapter, USB C with PD protocol, USB 5V or USB 20V...